tests/components/light/test_init.py

```diff
@@ -2,495 +2,623 @@
-        toggle_calls = mock_service(self.hass, light.DOMAIN, SERVICE_TOGGLE)
+    assert len(turn_on_calls) == 1
+    call = turn_on_calls[-1]
 
-        common.toggle(self.hass, entity_id="entity_id_val", transition="transition_val")
+    assert call.domain == light.DOMAIN
+    assert call.service == SERVICE_TURN_ON
+    assert call.data.get(ATTR_ENTITY_ID) == "entity_id_val"
+    assert call.data.get(light.ATTR_TRANSITION) == "transition_val"
+    assert call.data.get(light.ATTR_BRIGHTNESS) == "brightness_val"
+    assert call.data.get(light.ATTR_RGB_COLOR) == "rgb_color_val"
+    assert call.data.get(light.ATTR_XY_COLOR) == "xy_color_val"
+    assert call.data.get(light.ATTR_PROFILE) == "profile_val"
+    assert call.data.get(light.ATTR_COLOR_NAME) == "color_name_val"
+    assert call.data.get(light.ATTR_WHITE_VALUE) == "white_val"
 
-        self.hass.block_till_done()
+    # Test turn_off
+    turn_off_calls = async_mock_service(hass, light.DOMAIN, SERVICE_TURN_OFF)
 
-        assert 1 == len(toggle_calls)
-        call = toggle_calls[-1]
+    await hass.services.async_call(
+        light.DOMAIN,
+        SERVICE_TURN_OFF,
+        {
+            ATTR_ENTITY_ID: "entity_id_val",
+            light.ATTR_TRANSITION: "transition_val",
+        },
+        blocking=True,
+    )
 
-        assert light.DOMAIN == call.domain
-        assert SERVICE_TOGGLE == call.service
-        assert "entity_id_val" == call.data[ATTR_ENTITY_ID]
-        assert "transition_val" == call.data[light.ATTR_TRANSITION]
+    assert len(turn_off_calls) == 1
+    call = turn_off_calls[-1]
 
-    def test_services(self):
-        """Test the provided services."""
-        platform = getattr(self.hass.components, "test.light")
+    assert call.domain == light.DOMAIN
+    assert call.service == SERVICE_TURN_OFF
+    assert call.data[ATTR_ENTITY_ID] == "entity_id_val"
+    assert call.data[light.ATTR_TRANSITION] == "transition_val"
 
-        platform.init()
-        assert setup_component(
-            self.hass, light.DOMAIN, {light.DOMAIN: {CONF_PLATFORM: "test"}}
-        )
-        self.hass.block_till_done()
+    # Test toggle
+    toggle_calls = async_mock_service(hass, light.DOMAIN, SERVICE_TOGGLE)
 
-        ent1, ent2, ent3 = platform.ENTITIES
+    await hass.services.async_call(
+        light.DOMAIN,
+        SERVICE_TOGGLE,
+        {ATTR_ENTITY_ID: "entity_id_val", light.ATTR_TRANSITION: "transition_val"},
+        blocking=True,
+    )
 
-        # Test init
-        assert light.is_on(self.hass, ent1.entity_id)
-        assert not light.is_on(self.hass, ent2.entity_id)
-        assert not light.is_on(self.hass, ent3.entity_id)
+    assert len(toggle_calls) == 1
+    call = toggle_calls[-1]
 
-        # Test basic turn_on, turn_off, toggle services
-        common.turn_off(self.hass, entity_id=ent1.entity_id)
-        common.turn_on(self.hass, entity_id=ent2.entity_id)
+    assert call.domain == light.DOMAIN
+    assert call.service == SERVICE_TOGGLE
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6239266)